astropy/cosmology/_src/traits/curvature.py

```diff
@@ -0,0 +1,77 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Critical density.
+
+This is private API. See `~astropy.cosmology.traits` for public API.
+
+"""
+
+__all__ = ["CurvatureComponent"]
+
+import abc
+
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from astropy.cosmology._src.utils import aszarr, deprecated_keywords
+from astropy.units import Quantity
+
+
+class CurvatureComponent:
+    """The object has attributes and methods related to the global curvature."""
+
+    @property
+    @abc.abstractmethod
+    def Ok0(self) -> float | np.floating:
+        """Omega curvature; the effective curvature density/critical density at z=0."""
+        raise NotImplementedError
+
+    @deprecated_keywords("z", since="7.0")
+    def Ok(self, z: Quantity | ArrayLike) -> NDArray[np.floating] | np.floating:
+        """Return the equivalent density parameter for curvature at redshift ``z``.
+
+        Parameters
+        ----------
+        z : Quantity-like ['redshift'], array-like
+            Input redshift.
+
+            .. versionchanged:: 7.0
+                Passing z as a keyword argument is deprecated.
+
+        Returns
+        -------
+        Ok : ndarray
+            The equivalent density parameter for curvature at each redshift.
+
+            .. versionchanged:: 7.2
+                Always returns a numpy object, never a `float`.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from astropy.cosmology import Planck18, units as cu
+
+        >>> Planck18.Ok(2)
+        array(0.)
+
+        >>> Planck18.Ok([1, 2])
+        array([0., 0.])
+
+        >>> Planck18.Ok(np.array([2]))
+        array([0.])
+
+        >>> Planck18.Ok(2 * cu.redshift)
+        array(0.)
+
+        >>> cosmo = Planck18.clone(Ode0=0.71, to_nonflat=True)
+
+        >>> cosmo.Ok0  # doctest: +SKIP
```

Comment:
How do you plan to ensure these tests will be activated when support for `numpy` < 2 is dropped?

Reply:
@mhvk, seems workable.
I do wish this were as easy as https://sybil.readthedocs.io/en/latest/rest.html#skipping-examples for getting in-line solutions working. It's been a rough journey (https://github.com/astropy/astropy/pull/18232#discussion_r2171586888).

Reply:
Done. And it works. 😌

Reply:
Thanks!